src/DependencyInjection/Compiler/DoctrineSchemaPass.php

```diff
@@ -27,25 +26,14 @@ class DoctrineMigrationsPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$this->hasMigrationsBundle($container)) {
-            return;
-        }
-
         if ($this->hasOrm($container)) {
-            // Use Doctrine mapping (enhanced by our listeners) for schema if ORM is installed
-            $provider = new Definition(
-                'Doctrine\DBAL\Migrations\Provider\OrmSchemaProvider',
-                [$container->findDefinition('doctrine.orm.entity_manager')]
-            );
-        } else {
-            // Migrations schema provider must implement interface (only available if bundle is installed)
-            $provider = new DefinitionDecorator('contao.doctrine.dca_schema_provider');
-            $provider->setClass('Contao\CoreBundle\Doctrine\Schema\MigrationsSchemaProvider');
-
-            $this->registerDiffCommand($container, $provider);
+            $provider = $container->getDefinition('contao.doctrine.schema_provider');
+            $provider->addArgument($container->getDefinition('doctrine.orm.entity_manager'));
```

Comment:
Do not inject the entity manager here as it will result in circular reference:

```
  Circular reference detected for service "doctrine.dbal.default_connection", path: "cache_warmer -> contao.cache.warm_internal -> doctrine.dbal.default_connection -> contao.listener.doctrine_schema -> contao.doctrine.schema_provider -> doctrine.orm.defaults_entity_manager".  
```

Instead retrieve it at runtime in the `DcaSchemaProvider::createSchema()` at the very top.